Code/Framework/AzCore/AzCore/RTTI/BehaviorContextEBusEventRawSignature.inl

```diff
@@ -0,0 +1,230 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+#pragma once
+
+namespace AZ::Internal
+{
+    //////////////////////////////////////////////////////////////////////////
+    template<class EBus, BehaviorEventType EventType, class R, class BusType, class... Args>
+    BehaviorEBusEvent<EBus, EventType, R(BusType*, Args...)>::BehaviorEBusEvent(FunctionPointer functionPointer, BehaviorContext* context)
+        : BehaviorMethod(context)
+        , m_functionPtr(functionPointer)
+    {
+        SetParameters<R>(m_parameters, this);
+        SetParameters<Args...>(&m_parameters[s_startNamedArgumentIndex], this);
+        // optional ID parameter
+        SetBusIdType<s_isBusIdParameter == 1>();
+    }
+
+
+    //////////////////////////////////////////////////////////////////////////
+    template<class EBus, BehaviorEventType EventType, class R, class BusType, class... Args>
+    template<bool IsBusId>
```

Comment:
This template line now needs to be removed now that the `SetBusIdType` isn't a template function anymore

Reply:
Actually I'll just merge this code into the constructor. The more I look at it the more I realize that SetBusIdType() as a method is hurting readability, not helping it.